marsha/core/models.py

```diff
@@ -0,0 +1,392 @@
+    )
+
+    # `related_name` of fields pointing to this model, for typing
+    authoring: ReverseFKType["Authoring"]
+    playlists: ReverseFKType["Playlist"]
+
+    class Meta:
+        """Options for the ``Organization`` model."""
+
+        verbose_name: str = _("organization")
+        verbose_name_plural: str = _("organizations")
+
+
+class Authoring(BaseModel):
+    """Model representing authors in an organization."""
+
+    user: User = models.ForeignKey(
+        to=User,
+        related_name="authoring",
+        verbose_name=_("author"),
+        help_text=_("user having authoring access in this organization"),
+        # delete the user-organization link if the user is deleted
+        on_delete=models.CASCADE,
+    )
+    organization: Organization = models.ForeignKey(
+        to=Organization,
+        related_name="authoring",
+        verbose_name=_("organization"),
+        help_text=_("organization on which the user is an author"),
+        # delete the user-organization link if the organization is deleted
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        """Options for the ``Authoring`` model."""
+
+        verbose_name: str = _("author in organization")
+        verbose_name_plural: str = _("authors in organizations")
+        unique_together: UniqueTogether = [("user", "organization")]
+
+
+class Video(BaseModel):
+    """Model representing a video, by an author."""
+
+    author: User = models.ForeignKey(
+        to=User,
+        related_name="authored_videos",
+        verbose_name=_("author"),
+        help_text=_("author of the video"),
+        # don't delete a video if its author is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    video_language: str = models.CharField(
+        max_length=5,
+        verbose_name=_("video language"),
+        help_text=_("video language formatted like 'en_US'"),
+    )
+    audio_language: str = models.CharField(
+        max_length=5,
+        verbose_name=_("audio language"),
+        help_text=_("audio language formatted like 'en_US'"),
+    )
+    duplicated_from: "Video" = models.ForeignKey(
+        to="self",
+        related_name="duplicates",
+        verbose_name=_("duplicate from"),
+        help_text=_("original video this one was duplicated from"),
+        # don't delete a video if the one it was duplicated from is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+
+    # `related_name` of fields pointing to this model, for typing
+    audiotracks: ReverseFKType["AudioTrack"]
+    subtitletracks: ReverseFKType["SubtitleTrack"]
+    signtracks: ReverseFKType["SignTrack"]
+    duplicates: ReverseFKType["Video"]
+    playlists: M2MType["Playlist"]
+    playlists_links: ReverseFKType["PlaylistVideo"]
+
+    class Meta:
+        """Options for the ``Video`` model."""
+
+        verbose_name: str = _("video")
+        verbose_name_plural: str = _("videos")
+
+
+class VideoTrackBase(BaseModel):
```

Comment:
About the name. I like ends base classes with `Base`, like I do with `Mixin` for mixins. So `TrackBase`. So I agree to remove `Video`, renaming `VideoTrackBase` to `TrackBase`.

About `VideoTrack`, in #3 I talked about possibilities to have other kinds of video, asking for discussion about his.

> At this time we cannot think of another kind of attached video, so it's called ``SignsTrack``. If it happen in the future, it could be renamed ``AuxiliaryTrack``, with a ``kind`` field.

The ADR was accepted like this so... 

Do you think of any other kind of video that could be attached to a main video?

Reply:
I can't think of other kinds of video yet... so... ok to keep SignTrack.

For the Base, I have the feeling that everywhere Base is used as a prefix not a suffix. So I think we should follow this convention. First example is BaseModel from which you inherit here ;-)
Another example: https://github.com/divio/django-cms/blob/develop/cms/extensions/models.py
I'm sure there are many examples in Django's code base itself.

Also I feel it is clearer that `Base` is at the same place as what replaces it in children classes:
BaseTrack become (Video|Audio|Sign)Track.

Reply:
If you can't agree on this, maybe submit it to the vote on Slack ;-)

Reply:
Ok you convinced me!